diff --git a/soh/soh/OTRGlobals.cpp b/soh/soh/OTRGlobals.cpp
index b044c053..de2d8a0d 100644
--- a/soh/soh/OTRGlobals.cpp
+++ b/soh/soh/OTRGlobals.cpp
@@ -208,11 +208,11 @@ const char* constCameraStrings[] = {
 
 OTRGlobals::OTRGlobals() {
     std::vector<std::string> OTRFiles;
-    std::string mqPath = LUS::Context::GetPathRelativeToAppDirectory("oot-mq.otr");
+    std::string mqPath = LUS::Context::GetPathRelativeToAppBundle("oot-mq.otr");
     if (std::filesystem::exists(mqPath)) { 
         OTRFiles.push_back(mqPath);
     } 
-    std::string ootPath = LUS::Context::GetPathRelativeToAppDirectory("oot.otr");
+    std::string ootPath = LUS::Context::GetPathRelativeToAppBundle("oot.otr");
     if (std::filesystem::exists(ootPath)) {
         OTRFiles.push_back(ootPath);
     }
@@ -717,8 +717,8 @@ extern "C" void OTRExtScanner() {
 
 extern "C" void InitOTR() {
 #if not defined (__SWITCH__) && not defined(__WIIU__)
-    if (!std::filesystem::exists(LUS::Context::GetPathRelativeToAppDirectory("oot-mq.otr")) &&
-        !std::filesystem::exists(LUS::Context::GetPathRelativeToAppDirectory("oot.otr"))){
+    if (!std::filesystem::exists(LUS::Context::GetPathRelativeToAppBundle("oot-mq.otr")) &&
+        !std::filesystem::exists(LUS::Context::GetPathRelativeToAppBundle("oot.otr"))){
         bool generatedOtrIsMQ = false;
         if (Extractor::ShowYesNoBox("No OTR Files", "No OTR files found. Generate one now?") == IDYES) {
             Extractor extract;
