diff --git a/libultraship/src/Context.cpp b/libultraship/src/Context.cpp
index 6140e04..0396c8d 100644
--- a/libultraship/src/Context.cpp
+++ b/libultraship/src/Context.cpp
@@ -12,6 +12,8 @@
 #include "port/switch/SwitchImpl.h"
 #elif defined(__WIIU__)
 #include "port/wiiu/WiiUImpl.h"
+#elif defined(__linux__)
+#include <filesystem>
 #endif
 
 namespace LUS {
@@ -338,6 +340,20 @@ std::string Context::GetAppDirectoryPath(std::string appName) {
     if (fpath != NULL) {
         return std::string(fpath);
     }
+
+    std::filesystem::path dir;
+    fpath = std::getenv("XDG_DATA_HOME");
+    if (fpath != NULL) {
+        dir += fpath;
+	dir += "/soh";
+	return std::string(dir);
+    }
+    fpath = std::getenv("HOME");
+    if (fpath != NULL) {
+        dir += fpath;
+	dir += "/.local/share/soh";
+	return std::string(dir);
+    }
 #endif
 
 #ifdef NON_PORTABLE
